renderdoc/driver/metal/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+set(sources
+    official/metal-cpp.h
+    official/metal-cpp.cpp)
+
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
```

Comment:
For most people this requirement is likely satisfied already just by them being sufficiently up to date, so it's likely not an issue in practice, but C++17 is a higher compiler bar than other platforms so we should probably call this out explicitly in the Mac section of CONTRIBUTING/Compiling.md.